spec.html

```diff
@@ -37159,28 +37160,61 @@
+              1. If _strict_ is *false*, then
+                  1. For each |FunctionDeclaration| _f_ in _varDeclarations_ that is directly contained in the |StatementList| of a |Block|, |CaseClause|, or |DefaultClause|,
+                      1. Let _F_ be StringValue of the |BindingIdentifier| of |FunctionDeclaration| _f_.
+                      1. If replacing the |FunctionDeclaration| _f_ with a |VariableStatement| that has _F_ as a |BindingIdentifier| would not produce any Early Errors for _func_ and _F_ is not an element of BoundNames of _argumentsList_, then
+                          1. NOTE A var binding for _F_ is only instantiated here if it is neither a VarDeclaredName, the name of a formal parameter, or another |FunctionDeclaration|.
+                          1. If _instantiatedVarNames_ does not contain _F_, then
+                              1. Let _status_ be _varEnvRec_.CreateMutableBinding(_F_).
+                              1. Assert: _status_ is never an abrupt completion.
+                              1. Perform _varEnvRec_.InitializeBinding(_F_, *undefined*).
+                              1. Append _F_ to _instantiatedVarNames_.
+                          1. When the |FunctionDeclaration| _f_ is evaluated, perform the following steps in place of the |FunctionDeclaration| Evaluation algorithm provided in <emu-xref href="#sec-function-definitions-runtime-semantics-evaluation"></emu-xref>:
+                              1. Let _fenv_ be the running execution context's VariableEnvironment.
+                              1. Let _benv_ be the running execution context's LexicalEnvironment.
+                              1. Let _fobj_ be _benv_.GetBindingValue(_F_, *false*).
+                              1. Assert: _fobj_ is never an abrupt completion.
+                              1. Let _status_ be _fenv_.SetMutableBinding(_F_, _fobj_, *false*).
+                              1. Assert: _status_ is never an abrupt completion.
+                              1. Return NormalCompletion(~empty~).
+            </emu-alg>
+          </emu-annex>
+          <emu-annex id="sec-web-compat-globaldeclarationinstantiation">
+            <h1>Changes to GlobalDeclarationInstantiation</h1>
+            <p>During GlobalDeclarationInstatiation (<emu-xref href="#sec-globaldeclarationinstantiation"></emu-xref>) the following steps are performed in place of step 14:</p>
+            <emu-alg>
+              1. For each |FunctionDeclaration| _f_ that is directly contained in the |StatementList| of a |Block|, |CaseClause|, or |DefaultClause| Contained within script,
+                  1. Let _F_ be StringValue of the |BindingIdentifier| of |FunctionDeclaration| _f_.
+                  2. If replacing the |FunctionDeclaration| _f_ with a |VariableStatement| that has _F_ as a |BindingIdentifier| would not produce any Early Errors for script, then
+                      1. If _envRec_.HasLexicalDeclaration(_F_) is *false*, then
```

Comment:
I was considering cross-script errors and thought the error you get in that case is an early error, but I guess it's not because, while the error is thrown from ScriptEvaluation before any statements are evaluated, it's after any Early Errors are thrown. Thanks for the clarification!